### PR TITLE
Add ttl for sticky task list

### PR DESCRIPTION
**Pull request by vancexu**
This will delete sticky task list in one day after no ack update happens for that task list. 
Relate issue: https://github.com/uber/cadence/issues/472

### CODE REVIEW

**Comment by vancexu** [comment]
Integration test not done passed yet, need more work

File: common/persistence/cassandraPersistence.go [hunk 1]
```diff
@@ -1426,20 +1443,51 @@ func (d *cassandraPersistence) LeaseTaskList(request *LeaseTaskListRequest) (*Le
...
+	tli := &TaskListInfo{DomainID: request.DomainID, Name: request.TaskList, TaskType: request.TaskType, RangeID: rangeID + 1, AckLevel: ackLevel, Kind: request.TaskListKind}
 	return &LeaseTaskListResponse{TaskListInfo: tli}, nil
 }
 
 // From TaskManager interface
 func (d *cassandraPersistence) UpdateTaskList(request *UpdateTaskListRequest) (*UpdateTaskListResponse, error) {
 	tli := request.TaskListInfo
 
+	if tli.Kind == TaskListKindWorker { // if task_list is sticky, then update with TTL
+		fmt.Println("vancexu in updateTaskList with TTL")
```
Comment (yiminc) [inline common/persistence/cassandraPersistence.go]: remove

File: common/persistence/dataInterfaces.go [hunk 2]
```diff
@@ -59,6 +59,12 @@ const (
 	TaskListTypeActivity
 )
 
+// Kinds of task lists
+const (
+	TaskListKindNormal = iota
+	TaskListKindWorker
```
Comment (yiminc) [inline common/persistence/dataInterfaces.go]: rename to Sticky

File: idl/github.com/uber/cadence/shared.thrift [hunk 3]
```diff
@@ -196,6 +196,11 @@ enum HistoryEventFilterType {
   CLOSE_EVENT,
 }
 
+enum TaskListKind {
+  NORMAL,
+  WORKER,
```
Comment (yiminc) [inline idl/github.com/uber/cadence/shared.thrift]: rename WORKER to STICKY

File: schema/cadence/schema.cql [hunk 4]
```diff
@@ -132,6 +132,7 @@ CREATE TYPE task_list (
   name             text,
   type             int, -- enum TaskRowType {ActivityTask, DecisionTask}
   ack_level        bigint, -- task_id of the last acknowledged message
+  kind             int, -- enum TaskListKind {Normal, Worker}
```
Comment (yiminc) [inline schema/cadence/schema.cql]: update comments after rename WORKER to STICKY

File: common/persistence/cassandraPersistence.go (hunk 1 above)
Comment (vancexu) [reply common/persistence/cassandraPersistence.go]: Done, Thanks for catching this.

File: common/persistence/dataInterfaces.go (hunk 2 above)
Comment (vancexu) [reply common/persistence/dataInterfaces.go]: Done

**Comment by vancexu** [comment]
Manually test old client works, but in that case the tasklist will be treated as NORMAL all the time and wouldn't be removed automatically.

**Pull request merged**